crates/sui-core/src/authority.rs

```diff
@@ -3465,10 +3457,34 @@ impl AuthorityState {
             None
         };
 
+        let loaded_child_objects = if self.is_fullnode(epoch_store) {
+            // We only care about this for full nodes
+            inner_temporary_store.loaded_child_objects.clone()
+        } else {
+            BTreeMap::new()
+        };
+
         // Returns coin objects for indexing for fullnode if indexing is enabled.
         let tx_coins =
             self.fullnode_only_get_tx_coins_for_indexing(&inner_temporary_store, epoch_store);
 
+        // index certificate
+        let _ = self
+            .post_process_one_tx(
+                certificate,
+                effects,
+                &inner_temporary_store.events,
+                epoch_store,
+                tx_coins,
+                &inner_temporary_store.written,
```

Comment:
Now that you have changed the order, I believe you should just pass a reference to inner_temporary_store all the way down

Reply:
can you elaborate? passing inner_temp_store v.s. written/events?

Reply:
Right.

Reply:
why is the former better though?

Reply:
It would pass much fewer things around (which is the purpose of inner_temp_store). At the least we should get rid of the lines of creating  loaded_child_objects in line 3460, which is awkward (we can pass in reference and clone when we need it)